src/window/macos.rs

```diff
@@ -0,0 +1,138 @@
+use icrate::{
+    AppKit::{
+        NSColor, NSEvent, NSView, NSViewMinYMargin, NSViewWidthSizable, NSWindow,
+        NSWindowStyleMaskTitled,
+    },
+    Foundation::{MainThreadMarker, NSPoint, NSRect, NSSize},
+};
+use objc2::{declare_class, msg_send_id, mutability::InteriorMutable, rc::Id, ClassType};
+
+use csscolorparser::Color;
+use raw_window_handle::{HasRawWindowHandle, RawWindowHandle};
+use winit::window::Window;
+
+use crate::{cmd_line::CmdLineSettings, error_msg, frame::Frame, settings::SETTINGS};
+
+use super::WindowSettings;
+
+declare_class!(
+    // A view to simulate the double-click-to-zoom effect for `--frame transparency`.
+    struct TitlebarClickHandler;
+
+    unsafe impl ClassType for TitlebarClickHandler {
+        type Super = NSView;
+        type Mutability = InteriorMutable;
+        const NAME: &'static str = "TitlebarClickHandler";
+    }
+
+    unsafe impl TitlebarClickHandler {
+        #[method(mouseDown:)]
+        unsafe fn mouse_down(&self, event: &NSEvent) {
+            if event.clickCount() == 2 {
+                self.window().unwrap().zoom(Some(self));
+            }
+        }
+    }
+);
+
+impl TitlebarClickHandler {
+    pub fn new(_mtm: MainThreadMarker) -> Id<TitlebarClickHandler> {
+        unsafe { msg_send_id![Self::alloc(), init] }
+    }
+}
+
+lazy_static! {
+    // This height is in dpi-independent length, convert it to pixel length by multiply with scale factor.
+    static ref TITLEBAR_HEIGHT: f64 = MacosWindowFeature::calculate_titlebar_height();
+}
+
+pub struct MacosWindowFeature {
+    // Extra titlebar height in --frame transparecy. 0 in other cases.
+    extra_titlebar_height_in_pixel: u32,
+}
+
+impl MacosWindowFeature {
+    pub fn from_winit_window(window: &Window, mtm: MainThreadMarker) -> MacosWindowFeature {
+        let ns_window = if let RawWindowHandle::AppKit(handle) = window.raw_window_handle() {
+            unsafe { Id::retain(handle.ns_window as *mut NSWindow).unwrap() }
+        } else {
+            panic!("Not a appkit window.");
+        };
+
+        if let Ok(color) = &SETTINGS
+            .get::<WindowSettings>()
+            .background_color
+            .parse::<Color>()
+        {
+            error_msg!("neovide_background_color has now been deprecated. Use neovide_transparency instead if you want to get a transparent window titlebar. Please check https://neovide.dev/configuration.html#background-color-deprecated-currently-macos-only for more information.");
```

Comment:
Might make sense to split this over multiple lines? (idk if that exact suggestion works, might need to extract it into a variable)

```suggestion
            error_msg!(concat!(
                "neovide_background_color has now been deprecated. ",
                "Use neovide_transparency instead if you want to get a transparent window titlebar. ",
                "Please check https://neovide.dev/configuration.html#background-color-deprecated-currently-macos-only for more information.",
            ));
```